pkg/services/provisioning/dashboards/file_reader_test.go

```diff
@@ -148,6 +148,47 @@ func TestDashboardFileReader(t *testing.T) {
 				So(len(fakeService.inserted), ShouldEqual, 1)
 			})
 
+			Convey("Older dashboard file with the same checksum will not replace old dashboard", func() {
+				cfg.Options["path"] = oneDashboard
+				absPath, _ := filepath.Abs(oneDashboard + "/dashboard1.json")
+				stat, _ := os.Stat(oneDashboard + "/dashboard1.json")
+				file, _ := os.Open(absPath)
+				defer file.Close()
+
+				checksum, _ := util.Md5Sum(file)
+
+				fakeService.provisioned = map[string][]*models.DashboardProvisioning{
+					"Default": {
+						{Name: "Default", ExternalId: absPath, Updated: stat.ModTime().AddDate(0, 0, +1).Unix(), CheckSum: checksum},
+					},
+				}
+
+				reader, err := NewDashboardFileReader(cfg, logger)
+				So(err, ShouldBeNil)
+
+				err = reader.startWalkingDisk()
+				So(err, ShouldBeNil)
+				So(len(fakeService.inserted), ShouldEqual, 0)
+			})
+
+			Convey("Older dashboard file with different checksum will replace old dashboard", func() {
+				cfg.Options["path"] = oneDashboard
+				absPath, _ := filepath.Abs(oneDashboard + "/dashboard1.json")
+				stat, _ := os.Stat(oneDashboard + "/dashboard1.json")
+
+				fakeService.provisioned = map[string][]*models.DashboardProvisioning{
+					"Default": {
+						{Name: "Default", ExternalId: absPath, Updated: stat.ModTime().AddDate(0, 0, +1).Unix(), CheckSum: "fakechecksum"},
+					},
+				}
+
+				reader, err := NewDashboardFileReader(cfg, logger)
+				So(err, ShouldBeNil)
+
+				err = reader.startWalkingDisk()
+				So(err, ShouldBeNil)
+				So(len(fakeService.inserted), ShouldEqual, 1)
+			})
```

Comment:
Should be a blank line between tests.
```suggestion
			})

```